cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/CompositeLogAppender.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.appender;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Context;
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.logging.framework.local.LocalLogAppender;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A log appender that delegates logging events to a list of other log appenders.
+ */
+public class CompositeLogAppender extends LogAppender {
+
+  private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(CompositeLogAppender.class);
+  private final List<LogAppender> appenders;
+
+  /**
+   * Constructs a CompositeLogAppender with the given list of appenders.
+   *
+   * @param appenders the list of {@link LogAppender} instances to delegate logging events to
+   */
+  public CompositeLogAppender(List<LogAppender> appenders) {
+    if (appenders == null) {
+      throw new IllegalArgumentException("Appenders list cannot be null");
+    }
+
+    // Make the appenders list unmodifiable to ensure thread safety and prevent external
+    // modifications.
+    this.appenders = Collections.unmodifiableList(new ArrayList<>(appenders));
+    setName(getClass().getName());
+  }
+
+  @Override
+  public void start() {
```

Comment:
Is it ok to ignore the errors using `safelyExecute()` in case of start() method?

If some logAppender is not started, it can lead to undesirable state.

Reply:
+1. I consider logging an essential system. If it does not work it indicates serious misconfiguration. Continuing would lead to problems with error diagnostics.